### DISCUSSION TITLE
Updating a value (setValue) via watch

### DISCUSSION THREAD

**Discussion by davshoward** [react-hook-form/react-hook-form#6255] [Q&A]
I'm trying to update an input based on the value of it.

E.g. 4 checkboxes

1. value1
2. value2
3. value3
4. none

If 'none' is selected then I want to uncheck all the other buttons. I've been using the following:
```
const watchCheckboxes = watch('checkboxes');
useEffect(() => {
    const noneSelected = getValues(['checkboxes']).map((x) =>
	x.some((y) => y === 'none')
    )[0];

    noneSelected && setValue('checkboxes', ['none']);
}, [watchCheckboxes]);
```

But I get an infinite loop when selecting none. I've seen some other discussions/issues around this but I'm not clear on the correct approach to deal with this.

Any direction would be most appreciated :-)

**Comment by bluebill1049** [comment]
why not just move it out from `useEffect`, or doing it in `onChange`?

**Reply by davshoward** [reply]
Was initially as I couldn't get `onChange` to work, but realised I need to call it after `register`.

So that will do :-) Thanks!

**Reply by davshoward** [reply]
Saying that - it does get in the way and conflict with `register`...

I'm using a custom component like so:

```
const Checkbox = ({ name, label, value, onChange }) => {
    const { register } = useFormContext();

    return (
        <input
            type="checkbox"
	    name={name}
	    id={name}
	    value={value}
	    {...(register && register(name))}
	    onChange={onChange}
	/>
    );
};

export default Checkbox;
```
